api/app/Console/Kernel.php

```diff
@@ -15,7 +19,40 @@ class Kernel extends ConsoleKernel
      */
     protected function schedule(Schedule $schedule)
     {
-        // $schedule->command('inspire')->hourly();
+        $schedule->call(function () {
+            $deleteDate = Carbon::now()->subYears(5);
+            $users = User::onlyTrashed()->whereDate('deleted_at', '<=', $deleteDate)->get();
+
+            foreach ($users as $user) {
+                // Cascade delete to child models
+                foreach ($user->poolCandidates()->withTrashed()->get() as $candidate) {
+                    $candidate->forceDelete();
+                }
+                foreach ($user->awardExperiences()->withTrashed()->get() as $experience) {
+                    $experience->forceDelete();
+                }
+                foreach ($user->communityExperiences()->withTrashed()->get() as $experience) {
+                    $experience->forceDelete();
+                }
+                foreach ($user->educationExperiences()->withTrashed()->get() as $experience) {
+                    $experience->forceDelete();
+                }
+                foreach ($user->personalExperiences()->withTrashed()->get() as $experience) {
+                    $experience->forceDelete();
+                }
+                foreach ($user->workExperiences()->withTrashed()->get() as $experience) {
+                    $experience->forceDelete();
+                }
+                foreach ($user->roleAssignments()->get() as $role) {
+                    $role->forceDelete();
+                }
+                foreach ($user->userSkills()->withTrashed()->get() as $skill) {
+                    $skill->forceDelete();
+                }
+
+                $user->forceDelete();
+            }
+        })->daily();
```

Comment:
Not critical to the issue at hand, but I think scheduled functions should have their time set carefully and specifically. 
Ideally run at times of low server load, this execution should be fairly inexpensive as though it runs daily it would likely have zero people returned almost always and for sure for 5 years. But planning for possible further daily tasks like the other idea we floated with daily reporting #7082 

Laravel midnight would be 00:00 UTC or 20:00 EST or 17:00 PST so possibly executing these during possibly busy times. 
So I think we should establish a format for scheduled tasks to make it convenient to slot more in. 

Maybe `->dailyAt('08:00');` (so overnight in Canada)
And utilize `withoutOverlapping();` when adding further tasks

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/107acf20-237c-4a1a-9232-f8fb5d3daf56)

Reply:
Also, how do we find out if something goes wrong during a scheduled task?

Reply:
I've changed it to `->dailyAt('08:00');`. I don't think we need `withoutOverlapping` though since it looks like it prevents a single task from having multiple running instances at once which should be unnecessary in this case.

I'm not certain about server side but at least locally errors during a scheduled task show up in `storage/logs/laravel.log`. If that doesn't work on the server we could probably utilize [this](https://laravel.com/docs/10.x/scheduling#task-output).

Reply:
Nice stuff!
Yeah, the prevent overlapping would likely come into play for second and onward scheduled events